wallet/src/main/java/com/android/identity_credential/wallet/CardViewModel.kt

```diff
@@ -338,6 +338,9 @@ class CardViewModel : ViewModel() {
         secureAreaRepository: SecureAreaRepository,
         credentialTypeRepository: CredentialTypeRepository
     ) {
+        if (this::context.isInitialized) {
```

Comment:
It's in the description - `MainActivity.onCreate()` calls `setData()` and `MainActivity.onCreate()` is called on every configuration change. It's possible a better fix for this but the root problem is that we need to initialize `CardViewModel` and that's what `setData()` does...

Reply:
I think even as a temporary fix, this should be in onCreate and there should be a TODO to avoid this.

Longer term the work that we do when we stop observing (do we actually get onCleared call?) should probably undo what we do when we start observing? With lifecycle you kinda have to make sure all these paired calls do their paired work, so one undoes the other, otherwise there are still some issues in some weird scenarios.